### PR TITLE
Adding min and max yHeight/xWidth options for scroll bar handle

### PR DESCRIPTION
**Pull request by itsdrewmiller**
This is a feature addition not a bug, so I will not be offended if you don't want to use it.  I found that it was kind of frustrating when navigating across a long list of items to have the thumb of the scrollbar keep shrinking down to only a few pixels.

I updated the readme to include documentation for the options.  I could definitely see adding some defaults for this but I didn't want to break any existing functionality so I did not add them myself.

(Also I had no idea what the part was called - saw it called "thumb" here:  http://my.safaribooksonline.com/book/programming/microsoft-wpf/9780768695519/advanced-scrolling/ch07lev1sec1)

### CODE REVIEW

**Comment by itsdrewmiller** [comment]
Probably worth noting that I did NOT build the min stuff here - still gotta figure out how to do that.

File: src/perfect-scrollbar.js [hunk 1]
```diff
@@ -67,13 +69,23 @@
...
 
+      var getSettingsAdjustedThumbSize = function (thumbSize) {
+        if (settings.minThumbSize) {
+          thumbSize = Math.max(thumbSize, settings.minThumbSize);
+        }
+        if (settings.maxThumbSize) {
+          thumbSize = Math.min(thumbSize, settings.maxThumbSize);
+        }
+        return thumbSize;
+      }
```
Comment (hatashiro) [inline src/perfect-scrollbar.js]: var style function definition should finish with semicolon.

``` javascript
var functionName = function (...) {
  // ...
};
```

Please refer http://javascript.crockford.com/code.html.

File: src/perfect-scrollbar.js [hunk 2]
```diff
@@ -176,8 +188,8 @@
 
         $(document).bind('mousemove.perfect-scroll', function (e) {
           if ($scrollbarY.hasClass('in-scrolling')) {
-            updateContentScrollTop();
-            moveBarY(currentTop, e.pageY - currentPageY);
+             updateContentScrollTop();
```
Comment (hatashiro) [inline src/perfect-scrollbar.js]: why are these lines changed?

**Comment by hatashiro** [comment]
It looks very nice.

About the naming, I prefer to use just 'scrollbar' instead of 'thumb' and 'scrollbar' is now used in the codes.
And I think 'size' is kind of ambiguous because typically it means both width and height.
Also, I think 'max' option is not needed.

So, I suggest the option like:

``` javascript
{
  minScrollbarWidth: 100,
  minScrollbarHeight: 100
}
```

If you have some opinions, feel free to comment!

Thank you.

**Comment by GregDThomas** [comment]
> Probably worth noting that I did NOT build the min stuff here - still
> gotta figure out how to do that.

Has anyone given any thought to using node? It makes running tests and
creating min-ified files as simple as running a single command. I'm happy
to do the necessary refactoring if it's considered worthwhile.

Greg

**Comment by hatashiro** [comment]
I once considered to use node to build the scripts but I think it's kinda heavy job for just a jquery plugin.
So I just added build script and jshint file to use.
But I think it's right time to add a way to build and lint the codes.

I'm looking into this: http://gruntjs.com/

I'll do this in this weekend with new version release.

Thank you, Greg.

**Comment by GregDThomas** [comment]
> I'm looking into this: http://gruntjs.com/

Actually, that's exactly what I meant; I've got another plugin at
https://github.com/GregDThomas/jquery-localtime which uses grunt - you may
be able to use that as a starting point.

HTH,

Greg

**Comment by hatashiro** [comment]
@GregDThomas Thanks!

**Comment by hatashiro** [comment]
I adopted Grunt into this project.

Please refer https://github.com/noraesae/perfect-scrollbar/wiki/Contributing.

Thank you!

**Comment by itsdrewmiller** [comment]
I used "size" because it represents width for the X bar and height for the
Y bar.  :-)  but yeah it does seem ambiguous about if it also impacts the
other direction.  Maybe "length"?  With "scrollbar" that implies to me the
whole thing, like somehow the whole bar will only be 100px even if the
container is 400px.  I don't have a strong opinion though, and I can't
imagine anyone actually wanting the latter feature. I agree the max seems
like an option that maybe is a bad feature to have.  How about just:

{
    minScrollbarLength: 100,
}

File: src/perfect-scrollbar.js (hunk 1 above)
Comment (itsdrewmiller) [inline src/perfect-scrollbar.js]: Good call, I do in fact know this but my editor at work doesn't have jslint
or jshint installed :-)

File: src/perfect-scrollbar.js (hunk 2 above)
Comment (itsdrewmiller) [inline src/perfect-scrollbar.js]: I suspect this is my editor autotabbing to 4 instead of 2 - I'll correct
that.

**Comment by hatashiro** [comment]
Sounds good! :D

**Comment by itsdrewmiller** [comment]
Not sure if I am doing this right (this is like my third pull request and first one I am trying to update the code for) but I think the updates are now included in this - I corrected the spacing, missing ';' and changed the name and brought it down to only one setting. I also added an example page.

**Comment by hatashiro** [comment]
1. I can't automatically merge this pull request. Please rebase the branch with updated origin/master.
2. Please make all commits into 2 commits. one is for the minLength patch itself, and the other is for the example.
3. Please remove ',' and also lint the code with Grunt. About Grunt, please check https://github.com/noraesae/perfect-scrollbar/wiki/Contributing.

**Comment by ghost** [comment]
Is this change finished? Scrollbar is very small on large lists.

**Comment by fantasia949** [comment]
+1 for this feature
Hopefully, it'll be available soon :)

**Comment by itsdrewmiller** [comment]
I just deleted my old fork and redid the work on a new one - I still suck
at git but hopefully having just one commit makes things easier to merge.
:-) I followed the contributing guidelines, linted everything and so on so
hopefully there are no issues this time.

**Comment by itsdrewmiller** [comment]
I'm going to delete the old pull request (if I can) - new one is here:

https://github.com/noraesae/perfect-scrollbar/pull/48
